File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-shared-5-4/AverageDailyTradingVolumeLimit.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class AverageDailyTradingVolumeLimit : ISerialized
    {
        public AverageDailyTradingVolumeLimit(XmlNode xmlNode)
        {
            XmlNode limitationPercentageNode = xmlNode.SelectSingleNode("limitationPercentage");
            
            if (limitationPercentageNode != null)
            {
                if (limitationPercentageNode.Attributes["href"] != null || limitationPercentageNode.Attributes["id"] != null) 
                {
                    if (limitationPercentageNode.Attributes["id"] != null) 
                    {
                        limitationPercentageIDRef_ = limitationPercentageNode.Attributes["id"].Value;
                        RestrictedPercentage ob = new RestrictedPercentage(limitationPercentageNode);
                        IDManager.SetID(limitationPercentageIDRef_, ob);
                    }
                    else if (limitationPercentageNode.Attributes["href"] != null)
                    {
                        limitationPercentageIDRef_ = limitationPercentageNode.Attributes["href"].Value;
                    }
                    else
                    {
                        limitationPercentage_ = new RestrictedPercentage(limitationPercentageNode);
                    }
                }
                else
                {
                    limitationPercentage_ = new RestrictedPercentage(limitationPercentageNode);
                }
            }
            
        
            XmlNode limitationPeriodNode = xmlNode.SelectSingleNode("limitationPeriod");
            
            if (limitationPeriodNode != null)
            {
                if (limitationPeriodNode.Attributes["href"] != null || limitationPeriodNode.Attributes["id"] != null) 
                {
                    if (limitationPeriodNode.Attributes["id"] != null) 
                    {
                        limitationPeriodIDRef_ = limitationPeriodNode.Attributes["id"].Value;
                        XsdTypeNonNegativeInteger ob = new XsdTypeNonNegativeInteger(limitationPeriodNode);
                        IDManager.SetID(limitationPeriodIDRef_, ob);
                    }
                    else if (limitationPeriodNode.Attributes["href"] != null)
                    {
                        limitationPeriodIDRef_ = limitationPeriodNode.Attributes["href"].Value;
                    }
                    else
                    {
                        limitationPeriod_ = new XsdTypeNonNegativeInteger(limitationPeriodNode);
                    }
                }
                else
                {
                    limitationPeriod_ = new XsdTypeNonNegativeInteger(limitationPeriodNode);
                }
            }
            
        
        }
        
    
        #region LimitationPercentage_
        private RestrictedPercentage limitationPercentage_;
        public RestrictedPercentage LimitationPercentage_
        {
            get
            {
                if (this.limitationPercentage_ != null)
                {
                    return this.limitationPercentage_; 
                }
                else if (this.limitationPercentageIDRef_ != null)
                {
                    limitationPercentage_ = IDManager.getID(limitationPercentageIDRef_) as RestrictedPercentage;
                    return this.limitationPercentage_; 
                }
                else
                {
                      return this.limitationPercentage_; 
                }
            }
            set
            {
                if (this.limitationPercentage_ != value)
                {
                    this.limitationPercentage_ = value;
                }
            }
        }
        #endregion
        
        public string limitationPercentageIDRef_ { get; set; }
        #region LimitationPeriod_
        private XsdTypeNonNegativeInteger limitationPeriod_;
        public XsdTypeNonNegativeInteger LimitationPeriod_
        {
            get
            {
                if (this.limitationPeriod_ != null)
                {
                    return this.limitationPeriod_; 
                }
                else if (this.limitationPeriodIDRef_ != null)
                {
                    limitationPeriod_ = IDManager.getID(limitationPeriodIDRef_) as XsdTypeNonNegativeInteger;
                    return this.limitationPeriod_; 
                }
                else
                {
                      return this.limitationPeriod_; 
                }
            }
            set
            {
                if (this.limitationPeriod_ != value)
                {
                    this.limitationPeriod_ = value;
                }
            }
        }
        #endregion
        
        public string limitationPeriodIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-shared-5-4/AverageDailyTradingVolumeLimit.cpp
// AverageDailyTradingVolumeLimit.cpp 
#include "AverageDailyTradingVolumeLimit.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

AverageDailyTradingVolumeLimit::AverageDailyTradingVolumeLimit(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //limitationPercentageNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* limitationPercentageNode = xmlNode->FirstChildElement("limitationPercentage");

   if(limitationPercentageNode){limitationPercentageIsNull_ = false;}
   else{limitationPercentageIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- limitationPercentageNode , address : " << limitationPercentageNode << std::endl;
   #endif
   if(limitationPercentageNode)
   {
      if(limitationPercentageNode->Attribute("href") || limitationPercentageNode->Attribute("id"))
      {
          if(limitationPercentageNode->Attribute("id"))
          {
             limitationPercentageIDRef_ = limitationPercentageNode->Attribute("id");
             limitationPercentage_ = boost::shared_ptr<RestrictedPercentage>(new RestrictedPercentage(limitationPercentageNode));
             limitationPercentage_->setID(limitationPercentageIDRef_);
             IDManager::instance().setID(limitationPercentageIDRef_,limitationPercentage_);
          }
          else if(limitationPercentageNode->Attribute("href")) { limitationPercentageIDRef_ = limitationPercentageNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { limitationPercentage_ = boost::shared_ptr<RestrictedPercentage>(new RestrictedPercentage(limitationPercentageNode));}
   }

   //limitationPeriodNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* limitationPeriodNode = xmlNode->FirstChildElement("limitationPeriod");

   if(limitationPeriodNode){limitationPeriodIsNull_ = false;}
   else{limitationPeriodIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- limitationPeriodNode , address : " << limitationPeriodNode << std::endl;
   #endif
   if(limitationPeriodNode)
   {
      if(limitationPeriodNode->Attribute("href") || limitationPeriodNode->Attribute("id"))
      {
          if(limitationPeriodNode->Attribute("id"))
          {
             limitationPeriodIDRef_ = limitationPeriodNode->Attribute("id");
             limitationPeriod_ = boost::shared_ptr<XsdTypeNonNegativeInteger>(new XsdTypeNonNegativeInteger(limitationPeriodNode));
             limitationPeriod_->setID(limitationPeriodIDRef_);
             IDManager::instance().setID(limitationPeriodIDRef_,limitationPeriod_);
          }
          else if(limitationPeriodNode->Attribute("href")) { limitationPeriodIDRef_ = limitationPeriodNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { limitationPeriod_ = boost::shared_ptr<XsdTypeNonNegativeInteger>(new XsdTypeNonNegativeInteger(limitationPeriodNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<RestrictedPercentage> AverageDailyTradingVolumeLimit::getLimitationPercentage()
{
   if(!this->limitationPercentageIsNull_){
        if(limitationPercentageIDRef_ != ""){
             return boost::shared_static_cast<RestrictedPercentage>(IDManager::instance().getID(limitationPercentageIDRef_));
        }else{
             return this->limitationPercentage_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<RestrictedPercentage>();
   }
}
boost::shared_ptr<XsdTypeNonNegativeInteger> AverageDailyTradingVolumeLimit::getLimitationPeriod()
{
   if(!this->limitationPeriodIsNull_){
        if(limitationPeriodIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeNonNegativeInteger>(IDManager::instance().getID(limitationPeriodIDRef_));
        }else{
             return this->limitationPeriod_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeNonNegativeInteger>();
   }
}
}

